pkg/scraper/client/resource/decode_test.go

```diff
@@ -241,3 +242,27 @@ container_start_time_seconds{container="metrics-server",namespace="kubernetes-da
 		})
 	}
 }
+
+func Fuzz_decodeBatch(f *testing.F) {
+	f.Fuzz(func(t *testing.T, cpuValue float64, memValue float64, startTimeValue float64, timeStamp int64, defaultTimeValue int64, randomInput string, nodeName string) {
+		defaultTime := time.Unix(0, defaultTimeValue)
+		input := fmt.Sprintf(
+			`# HELP container_cpu_usage_seconds_total [ALPHA] Cumulative cpu time consumed by the container in core-seconds
+# TYPE container_cpu_usage_seconds_total counter
+container_cpu_usage_seconds_total{container="coredns",namespace="kube-system",pod="coredns-558bd4d5db-4dpjz"} %f %d
+# HELP container_memory_working_set_bytes [ALPHA] Current working set of the container in bytes
+# TYPE container_memory_working_set_bytes gauge
+container_memory_working_set_bytes{container="coredns",namespace="kube-system",pod="coredns-558bd4d5db-4dpjz"} %e %d
+# TYPE container_start_time_seconds gauge
+container_start_time_seconds{container="coredns",namespace="kube-system",pod="coredns-558bd4d5db-4dpjz"} %E %d`,
+			cpuValue, timeStamp, memValue, timeStamp, startTimeValue, timeStamp)
+		_, err := decodeBatch([]byte(input), defaultTime, "node1")
+		if err != nil && timeStamp >= 0 {
+			t.Errorf("Unexpect error: %v\nmetrics: %s\n", err, input)
+		}
+		_, err = decodeBatch([]byte(randomInput), defaultTime, nodeName)
```

Comment:
@serathius , I have added fuzzing with randomly generated data, do I need other scenarios?

Reply:
This should be moved to separate `f.Fuzz` call

Reply:
I haven't found a way to run multiple fuzz functions on one file. I will continue to study.

Reply:
Solution has been found.